torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -58,6 +58,56 @@ class IndexFlattener : public IRMutator {
   }
 };
 
+class ReductionExpander : public IRMutator {
```

Comment:
is this idempotent? if not can we add some assertions that it is run once

Reply:
Which case are you worried about? That it is run zero times or that it is run multiple times?